cmake/flash/CMakeLists.txt

```diff
@@ -1,93 +1,80 @@
 # SPDX-License-Identifier: Apache-2.0
 
+add_custom_target(flash_runner_property_target)
+
+function(write content)
+  set_property(
+    TARGET         flash_runner_property_target
+    APPEND_STRING
+    PROPERTY       yaml_contents
+    "${content}\n"
```

Comment:
I think we should let the user ensure the `\n`, so that he gets exactly what he expects.

especially, code like this below could end up having unexpected newlines:
(Personally I'm unsure what happens if you get an additional newline before `  - --gdb=${CMAKE_GDB}`)
```
 write("\
  - --board-dir=${BOARD_DIR}
  - --elf-file=${PROJECT_BINARY_DIR}/${KERNEL_ELF_NAME}
  - --hex-file=${PROJECT_BINARY_DIR}/${KERNEL_HEX_NAME}
  - --bin-file=${PROJECT_BINARY_DIR}/${KERNEL_BIN_NAME}")

  if (DEFINED CMAKE_GDB)
   write("  - --gdb=${CMAKE_GDB}")
  endif()
```

Reply:
IMO appending `\n` in the write-function makes the cmake code much more readable. 

The only reason the snippet you show is the way it is, is to show that the arguments are aligned.

Reply:
Either way is fine with me.

Reply:
Just tried it out, and it doesn't seems to harm, regardless of number of `\n`, so i'm fine with current approach.